src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -637,8 +650,12 @@ internal void SendMessage(Message message, ActivationData sendingActivation = nu
             // second, we check for a strategy associated with the target's interface. third, we check for a strategy associated with the activation sending the
             // message.
             var strategy = targetAddress.Grain.IsGrain ? catalog.GetGrainPlacementStrategy(targetAddress.Grain) : null;
+            var request = message.IsUsingIfaceVersion
+                ? message.GetDeserializedBody(this.serializationManager) as InvokeMethodRequest
+                : null;
+            var target = new PlacementTarget(message.TargetGrain, request?.InterfaceId ?? 0, request?.InterfaceVersion ?? 0);
```

Comment:
Struct seems like a good option to me.

---
In reply to: [107806329](https://github.com/dotnet/orleans/pull/2837#discussion_r107806329) [](ancestors = 107806329)